module/VuFind/src/VuFind/Search/EDS/Options.php

```diff
@@ -612,25 +694,16 @@ public function getAvailableExpanders()
      */
     protected function populateViewSettings()
     {
-        if (isset($this->apiInfo)
-            && isset($this->apiInfo['ViewResultSettings'])
-        ) {
-            //default result Limit
-            if (isset($this->apiInfo['ViewResultSettings']['ResultsPerPage'])) {
-                $this->defaultLimit
-                    = $this->apiInfo['ViewResultSettings']['ResultsPerPage'];
-            } else {
-                $this->defaultLimit = 20;
-            }
-
-            //default view (amount)
-            if (isset($this->apiInfo['ViewResultSettings']['ResultListView'])) {
-                $this->defaultView = 'list|'
-                    . $this->apiInfo['ViewResultSettings']['ResultListView'];
-            } else {
-                $this->defaultView = 'list|brief';
-            }
-        }
+        $settings = $this->apiInfo['ViewResultSettings'] ?? [];
+        // default result Limit
+        $this->defaultLimit = $this->defaultLimit
+            ?? $settings['ResultsPerPage']
+            ?? 20;
+
+        // default view
+        $this->defaultView = $this->defaultView
+            ?? $settings['ResultListView']
+            ?? 'list';
```

Comment:
Is there a reason this last fallback value has changed from 'list|brief' to just 'list'? I confess that I don't fully understand the purpose of the pipe-delimited default value if we only return the first section of it... but I didn't do a deep dive to see if the full version is used anywhere. Since we're still using pipes in some cases, I suspect we might want to keep them everywhere for consistency.

Reply:
Oops, fixed. That was accidentally left in while checking why the view icons don't display in EDS results. They still don't but the problem is unrelated.